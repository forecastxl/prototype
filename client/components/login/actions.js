import * as types from './actionTypes'

export const requestPasswordReset = payload => ({
  type: types.REQUEST_PASSWORD_RESET,
  payload
})

export const requestPasswordResetSuccess = payload => ({
  type: types.REQUEST_PASSWORD_RESET_SUCCESS,
  payload
})

export const requestPasswordResetFail = payload => ({
  type: types.REQUEST_PASSWORD_RESET_FAIL,
  payload
})

export const resetPassword = payload => ({
  type: types.RESET_PASSWORD,
  payload
})

export const resetPasswordSuccess = payload => ({
  type: types.RESET_PASSWORD_SUCCESS,
  payload
})
export const resetPasswordFail = payload => ({
  type: types.RESET_PASSWORD_FAIL,
  payload
})
