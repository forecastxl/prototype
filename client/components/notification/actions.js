import * as types from './actionTypes'

export const setNotification = notification => ({
  type: types.SET_NOTIFICATION,
  notification
})

export const unsetNotification = () => ({
  type: types.UNSET_NOTIFICATION
})
