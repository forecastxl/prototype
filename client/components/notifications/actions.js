import Notifications from 'react-notification-system-redux'

export const addNotification = ({ level, options }) => Notifications[level](options)
export const hideNotification = uid => Notifications.hide(uid)
