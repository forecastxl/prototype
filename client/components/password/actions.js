import { createAction } from 'redux-actions'
import * as types from './actionTypes'

export const requestResetPassword = createAction(types.REQUEST_RESET_PASSWORD)
export const requestResetPasswordSuccess = createAction(types.REQUEST_RESET_PASSWORD_SUCCESS)
export const requestResetPasswordFail = createAction(types.REQUEST_RESET_PASSWORD_FAIL)

export const resetPassword = createAction(types.RESET_PASSWORD)
export const resetPasswordSuccess = createAction(types.RESET_PASSWORD_SUCCESS)
export const resetPasswordFail = createAction(types.RESET_PASSWORD_FAIL)
