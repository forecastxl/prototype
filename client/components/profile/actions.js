import * as types from './actionTypes'

export const fetchProfile = () => ({
  type: types.FETCH_PROFILE,
  includeToken: true
})

export const fetchProfileSuccess = data => ({
  type: types.FETCH_PROFILE_SUCCESS,
  data
})

export const fetchProfileFail = errors => ({
  type: types.FETCH_PROFILE_FAIL,
  errors
})

export const updateProfile = user => ({
  type: types.UPDATE_PROFILE,
  includeToken: true,
  user
})

export const updateProfileSuccess = () => ({
  type: types.UPDATE_PROFILE_SUCCESS
})

export const updateProfileFail = errors => ({
  type: types.UPDATE_PROFILE_FAIL,
  errors
})
