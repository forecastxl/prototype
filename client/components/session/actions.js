import * as types from './actionTypes'

export const loginUser = payload => ({
  type: types.LOGIN_USER,
  payload
})

export const loginUserSuccess = payload => ({
  type: types.LOGIN_USER_SUCCESS,
  payload
})

export const loginUserFail = payload => ({
  type: types.LOGIN_USER_FAIL,
  payload
})

export const logoutUser = () => ({
  type: types.LOGOUT_USER
})
