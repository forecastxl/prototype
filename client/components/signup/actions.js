import * as types from './actionTypes'

export const requestAccount = payload => ({
  type: types.REQUEST_ACCOUNT,
  payload
})

export const requestAccountSuccess = payload => ({
  type: types.REQUEST_ACCOUNT_SUCCESS,
  payload
})

export const requestAccountFail = payload => ({
  type: types.REQUEST_ACCOUNT_FAIL,
  payload
})
