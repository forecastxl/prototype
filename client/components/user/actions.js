import * as types from './actionTypes'

// confirm newly created account with a token sent via email

export const confirmAccount = token => ({
  type: types.CONFIRM_ACCOUNT,
  api: { key: types.CONFIRM_ACCOUNT, isFetching: true, errors: {} },
  token
})

export const confirmAccountSuccess = token => ({
  type: types.CONFIRM_ACCOUNT_SUCCESS,
  api: { key: types.CONFIRM_ACCOUNT, isFetching: false, errors: {} },
  token
})

export const confirmAccountFail = errors => ({
  type: types.CONFIRM_ACCOUNT_FAIL,
  api: { key: types.CONFIRM_ACCOUNT, isFetching: false, errors }
})

// create a new account with data from a form

export const createAccount = account => ({
  type: types.CREATE_ACCOUNT,
  api: { key: types.CREATE_ACCOUNT, isFetching: true, errors: {} },
  account
})

export const createAccountSuccess = () => ({
  type: types.CREATE_ACCOUNT_SUCCESS,
  api: { key: types.CREATE_ACCOUNT, isFetching: false, errors: {} }
})

export const createAccountFail = errors => ({
  type: types.CREATE_ACCOUNT_FAIL,
  api: { key: types.CREATE_ACCOUNT, isFetching: false, errors }
})

// request a token with userdata

export const login = user => ({
  type: types.LOGIN,
  api: { key: types.LOGIN, isFetching: true, errors: {} },
  user
})

export const loginSuccess = token => ({
  type: types.LOGIN_SUCCESS,
  api: { key: types.LOGIN, isFetching: false, errors: {} },
  token
})

export const loginFail = errors => ({
  type: types.LOGIN_FAIL,
  api: { key: types.LOGIN, isFetching: false, errors }
})

// destroy the current token

export const logout = () => ({
  type: types.LOGOUT
})

// request a password change, for users who forgot it

export const requestResetPassword = email => ({
  type: types.REQUEST_RESET_PASSWORD,
  api: { key: types.REQUEST_RESET_PASSWORD, isFetching: true, errors: {} },
  email
})

export const requestResetPasswordSuccess = () => ({
  type: types.REQUEST_RESET_PASSWORD_SUCCESS,
  api: { key: types.REQUEST_RESET_PASSWORD, isFetching: false, errors: {} }
})

export const requestResetPasswordFail = errors => ({
  type: types.REQUEST_RESET_PASSWORD_FAIL,
  api: { key: types.REQUEST_RESET_PASSWORD, isFetching: false, errors }
})

// change the existing password and login the user on success

export const resetPassword = payload => ({
  type: types.RESET_PASSWORD,
  api: { key: types.RESET_PASSWORD, isFetching: true, errors: {} },
  payload
})

export const resetPasswordSuccess = token => ({
  type: types.RESET_PASSWORD_SUCCESS,
  api: { key: types.RESET_PASSWORD, isFetching: false, errors: {} },
  token
})

export const resetPasswordFail = errors => ({
  type: types.RESET_PASSWORD_FAIL,
  api: { key: types.RESET_PASSWORD, isFetching: false, errors }
})
