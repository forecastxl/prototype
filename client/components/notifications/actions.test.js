import * as actions from './actions'

describe('actioncreators', () => {
  it('should create a RNS_SHOW_NOTIFICATION action', () => {
    const level = 'success'
    const options = { title: 'title', message: 'message', uid: 1 }
    expect(actions.addNotification({ level, options })).toMatchSnapshot()
  })

  it('should create a RNS_HIDE_NOTIFICATION action', () => {
    expect(actions.hideNotification(1)).toMatchSnapshot()
  })
})
