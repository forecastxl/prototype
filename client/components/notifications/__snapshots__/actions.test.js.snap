// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actioncreators should create a RNS_HIDE_NOTIFICATION action 1`] = `
Object {
  "type": "RNS_HIDE_NOTIFICATION",
  "uid": 1,
}
`;

exports[`actioncreators should create a RNS_SHOW_NOTIFICATION action 1`] = `
Object {
  "level": "success",
  "message": "message",
  "title": "title",
  "type": "RNS_SHOW_NOTIFICATION",
  "uid": 1,
}
`;
